src/zenml/integrations/mlflow/model_registries/mlflow_model_registry.py

```diff
@@ -0,0 +1,796 @@
+            filter_string=filter_string,
+            max_results=100,
+        )
+        # Return the registered models.
+        return [
+            ModelRegistration(
+                name=registered_model.name,
+                description=registered_model.description,
+                tags=registered_model.tags,
+            )
+            for registered_model in registered_models
+        ]
+
+    def check_model_exists(
+        self,
+        name: str,
+    ) -> bool:
+        """Check if a model exists in the MLflow model registry.
+
+        Args:
+            name (str): The name of the model.
+
+        Returns:
+            True if the model exists, False otherwise.
+        """
+        # Check if the model exists.
+        try:
+            self.mlflow_client.get_registered_model(name=name)
+        except MlflowException:
+            return False
+        return True
+
+    # ---------
+    # Model Version Methods
+    # ---------
+
+    def register_model_version(
+        self,
+        name: str,
+        description: Optional[str] = None,
+        tags: Optional[Dict[str, str]] = None,
+        model_source_uri: Optional[str] = None,
+        version: Optional[str] = None,
+        version_description: Optional[str] = None,
+        version_tags: Optional[Dict[str, str]] = None,
+        registry_metadata: Optional[Dict[str, str]] = None,
+        zenml_version: Optional[str] = None,
+        zenml_pipeline_run_id: Optional[str] = None,
+        zenml_pipeline_name: Optional[str] = None,
+        zenml_step_name: Optional[str] = None,
+        **kwargs: Any,
+    ) -> ModelVersion:
+        """Register a model version to the MLflow model registry.
+
+        Args:
+            name (str): The name of the model.
+            description (str, optional): The description of the model.
+            tags (Dict, optional): The tags of the model.
+            model_source_uri (str, optional): The source URI of the model.
+            version (str, optional): The version of the model.
+            version_description (str, optional): The description of the model
+                version.
+            version_tags (Dict, optional): The tags of the model version.
+            registry_metadata (Dict, optional): The registry metadata of the
+                model version.
+            zenml_version (str, optional): The ZenML version.
+            zenml_pipeline_run_id (str, optional): The ZenML pipeline run ID.
+            zenml_pipeline_name (str, optional): The ZenML pipeline name.
+            zenml_step_name (str, optional): The ZenML step name.
+            **kwargs: Additional keyword arguments.
+
+        Raises:
+            MlflowException: If the registered model does not exist.
+
+        Returns:
+            The registered model version.
+        """
+        # Check if the model exists, if not create it.
+        if not self.check_model_exists(name):
+            logger.warning(
```

Comment:
Is this really worthy of a warning, I thought this was sort of intended behaviour that we automatically register the model if creating the first version?

Reply:
I think it does, the default behavior of the mlflow client here would be to raise an error, so logging this as `info` might be good.